File: 1. Two Sum/twoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        final int[] results = new int[2];
        final Map<Integer, Integer> hash = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            hash.put(nums[i], i);
        }
        
        for (int i = 0; i < nums.length; i++) {
            if (hash.containsKey(target - nums[i])) {
                int index = hash.get(target - nums[i]);
                if (index == i) {
                    continue;
                }
                results[0] = i;
                results[1] = index;
                break;
            }
        }
        return results;
    }
}

File: 1. Two Sum/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
            :type nums: List[int]
            :type target: int
            :rtype: List[int]
            """
        hash = {}
        for i in range(0, len(nums)):
            hash[target - nums[i]] = i
        
        for i in range(0, len(nums)):
            if nums[i] in hash and hash[nums[i]] != i:
                return [hash[nums[i]], i]

